### DISCUSSION TITLE
Allow EXR Image files for HDR Envmaps lightmaps converted to ktx2

### DISCUSSION THREAD

**Discussion by arpu** [zeux/meshoptimizer#580] [General]
Hey,
looking at possible use for and gltfpack https://github.com/KhronosGroup/glTF/tree/KHR_lights_environment/extensions/2.0/Khronos/KHR_lights_environment

KTX-Software ktx create v4.3.0 now allows convert from EXR https://github.com/KhronosGroup/KTX-Software/discussions/696

is this something you plan to implement?

**Comment by zeux** [comment]
This would need EXR support in https://github.com/BinomialLLC/basis_universal I think.

**Reply by zeux** [reply]
Hmm or maybe not. I need to look at the extension more closely.

**Reply by zeux** [reply]
Oh, ok, I don't think this would require support in external software - it would just require retaining KTX2 images as the extension specifies their use. However, note that this extension is *not* currently part of glTF spec (the link you provided is to a branch with an open PR). If and when this extension gets finalized and merged to glTF mainline then yeah gltfpack will support it.

**Reply by arpu** [reply]
oh right https://github.com/KhronosGroup/glTF/pull/1956

**Reply by meshula** [reply]
@zeux I don't see the branch with the open PR you mention ~ is it a proposal for KTX2 to encapsulate EXR images? I'm currently looking at EXR and KTX2 support in USDZ as part of the glTF/USD interop effort, and it would be nifty to hit both at once.

**Reply by zeux** [reply]
@meshula I think EXR is mostly unrelated/separate here:

- There's a glTF PR, linked above, which argues for adding an image-based lighting extension that is using KTX2 cubemap as a source of HDR data;
- There's support in KTX-Software to produce KTX2 cubemaps with HDR data from EXR files if I understand correctly.

**Reply by meshula** [reply]
Thanks :) I got maybe a little excited at some possible convergence there. I'm knee-deep in exr code at the moment, we've reworked the core code to be a ton faster, and aras helped us tune compression so yeah, that's me, jumping up and down and waving ;)

I found the ktx create tool and the exr conversion. It could go further than that tool does though, EXR has got cube and latlong maps, but the tool only processes basic scanline images.
